src/components/Confirm/Confirm.js

```diff
@@ -219,15 +229,104 @@ class Confirm extends Component {
         sdkMetadata,
         ...issuingCountry,
       }
-      uploadDocument(data, url, token, this.onApiSuccess, this.onApiError)
+      if (useDecoupleCallbacks)
+        this.onSubmitCallback(data, CALLBACK_TYPES.document)
+      else uploadDocument(data, url, token, this.onApiSuccess, this.onApiError)
     } else if (method === 'face') {
       if (variant === 'video') {
         const data = { challengeData, blob, language, sdkMetadata }
-        uploadLiveVideo(data, url, token, this.onApiSuccess, this.onApiError)
-      } else {
-        this.handleSelfieUpload(capture, token)
+        if (useDecoupleCallbacks)
+          this.onSubmitCallback(data, CALLBACK_TYPES.video)
+        else
+          uploadLiveVideo(data, url, token, this.onApiSuccess, this.onApiError)
+      } else if (useDecoupleCallbacks)
+        this.onSubmitCallback(capture, CALLBACK_TYPES.selfie)
+      else this.handleSelfieUpload(capture, token)
+    }
+  }
+
+  onSubmitCallback = (data, callbackName) => {
```

Comment:
I'm open to suggestions for improving this function. I've tried multiple iterations and still not super happy with it.

Reply:
Yeah, there's that, and I'm not sure if I like having to do all the conditional code. An alternative I considered was separating this into 3 functions, one for each callback. The problem with that is that it's more code and it's mostly repeated code with the only real difference between them being the callbacks/functions being invoked and the payloads.

Reply:
Thanks for the feedback, love the suggestion and have made the change.